src/raftstore/store/config.rs

```diff
@@ -18,6 +18,61 @@ lazy_static! {
     .unwrap();
 }
 
+/// Custom quorum function for a Raft node.
+#[derive(Clone, Debug, PartialEq)]
+pub enum QuorumAlgorithm {
+    /// Default quorum function described in Raft paper.
+    Majority,
+    /// Ensure no data lost when ceil(voters_len / 2) voters fail.
+    IntegrationOnHalfFail,
+}
+
+mod quorum_algorithm_serde {
```

Comment:
Why not just derive `Serialize` and `Deserialize`? It should be similar to `engine::rocks::util::config::CompressionType`.